### ISSUE TITLE
Prediction with categorical variable and exog=Series

### ISSUE THREAD

**Issue by kshedden** [statsmodels/statsmodels#1881]
I'm trying to use predict with an OLS model fit from a formula in which one of the variables is a categorical type, and the exog passed to predict is a Series.

The code below raises a PatsyError: categorical data must be an iterable container.

If there is no categorical in the formula, I can pass either a Series or a DataFrame with no problem.  If there is a categorical in the formula, I can pass a DataFrame (even with a single row), but not a Series.

This is Pandas 0.14.1 and Statsmodels 0.5.0

```
import statsmodels.api as sm
import pandas as pd
data = np.random.normal(size=(10,4))
data[:,1] = np.random.randint(0, 4, 10)
data = pd.DataFrame(data, columns=["y", "a", "b", "c"])
mod = sm.OLS.from_formula("y ~ C(a) + b + c", data)
rslt = mod.fit()

# This works
rslt.predict(exog=data.iloc[0:1,:])


# This does not work
rslt.predict(exog=data.iloc[0,:])
```

I'm not sure if this is related to #1342.

**Comment by josef-pkt** [comment]
Temporary prio-high to check and close

**Comment by josef-pkt** [comment]
fixed with patsy 0.4
